services/printers/paperboy/paperboy/printer.py

```
@@ -19,19 +19,41 @@ class Printer:
         "Apathy": "😶 @ Lounge",
     }
 
+    PRINTER_HAS_COLOR = {
+        "Love": True,
+        "Hope": True,
+        "Joy": True,
+        "Peace": False,
+        "Apathy": False,
+    }
+
     def get_short_id(self) -> str:
         if self.name in self.PRINTER_ALIAS:
             return self.PRINTER_ALIAS[self.name]
         return self.get_id()
 
     def get_id(self) -> str:
         return f"{self.name} @ {self.location}"
+    
+    def get_cap_color(self) -> bool:
+        if self.name in self.PRINTER_HAS_COLOR:
+            return self.PRINTER_HAS_COLOR[self.name]
+        else:
+            # Hopefully this path never gets used!
+            conn = cups.Connection()
+            color_modes: (list[str]) = conn.getPrinterAttributes(self.name, requested_attributes=["print-color-mode-supported"]).get("print-color-mode-supported")
+            # It *shouldn't* ever return something other than a list[str] but...
+            if type(color_modes) is not list[str]:
+                return False
```

Might it make more sense to `raise TypeError` (I forget the exact syntax for python) rather than failing silently?

---

I worry this would break the bot completely? (Granted if CUPS is returning bad data then it probably isn't of much use anyway...)

---

Fair, at the very least it should log a warning though